docs/content/topics/operators/gateways.md

```diff
@@ -11,340 +11,421 @@ aliases:
 ```
 
-Again, note that any fields in the `data:` section above are shown cleartext,
-though in reality you _must_ base-64 encode them.
-
-The easiest way to create a secret like the above is to do so with the `kubectl`
-command, though there are a host of language-specific libraries now for creating
-secrets in code.
+Let's look at the high-level sections in the event definition above.  They are:
+
+  1. Event metadata, including:
+    i. The `apiVersion` of the schema for this event
+    ii. The schema `kind`, which will always be `Event`
+    iii. The `id` of the Event
+    iv. A `created` timestamp for the Event
+    v. The `projectID` that the Event is associated with
+    vi. The event `source`
+    vii. The event `type`
+  2. The `worker.spec` section, which contains worker configuration inherited
+    from the project definition associated with the event in combination with
+    system-level defaults.
+  3. The `worker.status` section, which contains the `started` and `ended`
+    timestamps and current `phase` of the worker handling the event. In the
+    example above, it has already reached the terminal phase of `SUCCEEDED`.
+
+To explore the SDK definitions of an Event object, see the [Go SDK Event] and
+[JavaScript/TypeScript SDK Event].
+
+[01-hello-world sample project]: https://github.com/brigadecore/brigade/tree/v2/examples/01-hello-world
+[Go SDK Event]: https://github.com/brigadecore/brigade/blob/v2/sdk/v2/core/events.go
+[JavaScript/TypeScript SDK Event]: https://github.com/brigadecore/brigade-sdk-for-js/blob/master/src/core/events.ts
 
 ## Creating Custom Gateways
 
-Given the above description of how gateways work, we can now talk about a gateway
-as anything that generates a secret following the format above.
-
-In this final section, we will create a simple shell script that triggers a new
-event every 60 seconds. In the payload, it sends the system time of the host
-that is running the script.
-
-```bash
-#!/usr/bin/env bash
-set -euo pipefail
-
-# The Kubernetes namespace in which Brigade is running.
-namespace="default"
-
-event_provider="simple-event"
-event_type="my_event"
-
-# This is github.com/brigadecore/empty-testbed
-project_id="brigade-830c16d4aaf6f5490937ad719afd8490a5bcbef064d397411043ac"
-commit_ref="master"
-commit_id="589e15029e1e44dee48de4800daf1f78e64287c0"
-
-base64=(base64)
-uuidgen=(uuidgen)
-if [[ "$(uname)" != "Darwin" ]]; then
-  base64+=(-w 0)
-  uuidgen+=(-t) # generate UUID v1 for sortability
-fi
-
-# This is the brigade script to execute
-script=$(cat <<EOF
-const { events } = require("brigadier");
-events.on("my_event", (e) => {
-  console.log("The system time is " + e.payload);
-});
-EOF
-)
+Given the above description of how gateways work, we can walk through how a
+minimal example gateway can be built. We'll focus on the event creation side
+of a Brigade gateway, rather than going over other common attributes, such as
+an http(s) server that awaits external webhook events.
```

Comment:
Nit: Elsewhere in this doc, http(s) is capitalized. I don't have a preference for upper/lowercase, but we should be internally consistent.